modules/hitl2/src/backend/repository.ts

```diff
@@ -0,0 +1,367 @@
+import axios from 'axios'
+import * as sdk from 'botpress/sdk'
+import Knex from 'knex'
+import _ from 'lodash'
+
+import { AgentType, CommentType, EscalationType } from './../types'
+import { makeAgentId } from './helpers'
+
+export interface AgentCollectionConditions {
+  online?: boolean
+}
+
+export interface CollectionConditions {
+  limit?: number
+  orderByColumn?: string
+  orderByDirection?: 'asc' | 'desc'
+}
+
+export default class Repository {
+  private escalationColumns: string[]
+  private commentColumns: string[]
+  private eventColumns: string[]
+  private commentColumnsPrefixed: string[]
+  private commentPrefix: string
+  private escalationPrefix: string
+
+  constructor(private bp: typeof sdk) {
+    this.commentPrefix = 'comment'
+    this.escalationPrefix = 'escalation'
+
+    this.escalationColumns = [
+      'id',
+      'botId',
+      'agentId',
+      'userId',
+      'userThreadId',
+      'agentThreadId',
+      'status',
+      'assignedAt',
+      'resolvedAt',
+      'createdAt',
+      'updatedAt'
+    ]
+
+    this.commentColumns = ['id', 'agentId', 'content', 'createdAt', 'updatedAt']
+
+    this.eventColumns = ['id', 'direction', 'botId', 'channel', 'success', 'createdOn', 'threadId', 'event']
+
+    this.commentColumnsPrefixed = this.commentColumns.map(s => this.commentPrefix.concat(':', s))
+  }
+
+  private axiosConfig = async (req, botId: string): Promise<sdk.AxiosBotConfig> => {
+    const config = await this.bp.http.getAxiosConfigForBot(botId, { localUrl: true })
+    config.baseURL = _.trim(config.baseURL, `/bots/${botId}`)
+    config.headers['Authorization'] = req.headers.authorization
+    config.headers['X-BP-Workspace'] = req.workspace
+    return config
```

Comment:
getAxiosConfigForBot is specifically to get a temporary authorization for a module. Since you want to use the user's authorization token, you can avoid this step. Ex: 
```js
return {
  baseURL: `${process.LOCAL_URL}/api/v1/bots/${botId}`,
  headers: {
    Authorization: req.headers.authorization,
    'X-BP-Workspace: req.workspace
}
```